src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -140,6 +142,10 @@ trait Blockchain {
 
   def saveNode(nodeHash: NodeHash, nodeEncoded: NodeEncoded, blockNumber: BigInt): Unit
 
+  def saveBlockNumber(number: BigInt, hash: ByteString): Unit
+
+  def saveBestBlockNumber(number: BigInt): Unit
```

Comment:
I wonder if we really need this method? We already have `saveBlockNumber`, maybe it could also update best block num automatically?